tests/test_completion.py

```diff
@@ -0,0 +1,197 @@
+import pytest
+import shlex
+from time import sleep
+
+commands_without_input = shlex.split(
+    """
+        quit true false reload version
+        list_commands list_monitors
+        list_rules list_keybinds
+        lock unlock
+        close_or_remove close_and_remove
+        remove rotate
+        mouseunbind
+        use_previous
+        lock unlock
+    """)
+
+
+def command2str(command):
+    s = ' '.join(command)
+    maxlen = 20
+    if len(s) > maxlen:
+        s = s[0:maxlen - 3] + '...'
```

Comment:
I suggest `s = s[0:maxlen - 1] + '…'`

Reply:
I had this originally, but unfortunately pytest prints unicode characters quoted:
```
test_my_test_case[some_long_para\u2026]
```
Currently, this helper function is unused anyway so I've commented this out for the moment.

Reply:
Doh! Looks like they are totally weird about this as well (https://github.com/pytest-dev/pytest/issues/2482)

Reply:
Bummer. I didn't even know we had an inside man there.